### ISSUE TITLE
Would it be possible to make the retryInterval an exponential delay like for example retryInterval=retryInterval*1.6

### ISSUE THREAD

**Issue by ymedaghri** [andywer/pg-listen#41]
Hi,
A lot a tools uses an exponential factor for retries (PM2, ....), could it be possible to change the retryInterval from a number to a function or to an exponential retryInterval ?
Thanks

**Comment by andywer** [comment]
Hey @ymedaghri!

Yes, I thought about the ability to pass a callback myself already, but postponed that thought until someone has an actual need for that. So here we are now… 😉

The question is just how the callback API should look like. Maybe the `retryInterval` could look like this:

```ts
/**
 * @param attempt No. of consecutive connection attempts that failed already.
 * @return Delay to wait until we try again in milliseconds.
 */
type RetryIntervalCallback = (attempt: number) => number
```

What do you think?

**Comment by andywer** [comment]
PS: The main reason why it's not an exponential delay in the first place is that this connection is critical to most services that use it and the re-connection attempt is cheap, so my thinking was that I'd rather try to reconnect to often than the functionality not working for a while after the database connection is back up again.

**Comment by ymedaghri** [comment]
Hi Andy,

Thank you for your quick answer, I'm impressed, I am not used to such quick answers :)
Yes the RetryIntervalCallback you are proposing are a perfect fit for our needs, since our queue is dequeuing DDD domainEvents as a background task so this is not extremely critical for us (we use eventual consistency).

We were previously using pg-pubsub but we had inexpected behavior on production since we lost the persistent connection and we were unable to detect it from our error logs.
I found your library and since you were mentioning that you created this library due to production reliability issue with other libraries, that was a very good point for us.
 
Thanks a lot for your hard work and this library,
Best Regards,
